library/x509write_crt.c

```diff
@@ -178,10 +188,24 @@ int mbedtls_x509write_crt_set_subject_key_identifier( mbedtls_x509write_cert *ct
     MBEDTLS_ASN1_CHK_ADD( len,
                 mbedtls_pk_write_pubkey( &c, buf, ctx->subject_key ) );
 
+#if defined(MBEDTLS_USE_PSA_CRYPTO)
+    status = psa_hash_compute( PSA_ALG_SHA_1,
+                               buf + sizeof( buf ) - len,
+                               len,
+                               buf + sizeof( buf ) - 20 ,
+                               PSA_HASH_LENGTH(PSA_ALG_SHA_1),
+                               &hash_length );
+    if( status != PSA_SUCCESS )
+    {
+        return( MBEDTLS_ERR_ERROR_GENERIC_ERROR );
+    }
```

Comment:
`MBEDTLS_ERR_PLATFORM_HW_ACCEL_FAILED` would make more sense as a generic error to indicate “crypto failed and I don't know why”. In general we should have a function to do this kind of error translation, but that's out of scope here, I've filed I've filed https://github.com/ARMmbed/mbedtls/issues/5514 .
```suggestion
        return( MBEDTLS_ERR_ERROR_HW_ACCEL_FAILED );
```